packages/data/addon/services/navi-metadata-apollo.js

```diff
@@ -0,0 +1,76 @@
+import ApolloService from 'ember-apollo-client/services/apollo';
+import { setContext } from 'apollo-link-context';
+import { getHost } from 'navi-data/utils/adapter';
+
+export default class NaviMetadataApolloService extends ApolloService {
+  /**
+   * @property {String}
+   */
+  namespace = 'v1/graphql';
+
+  /**
+   * @override
+   * @method clientOptions
+   * @returns {Object}
+   */
+  clientOptions() {
+    return {
+      link: this.link(),
+      cache: this.cache(),
+      defaultOptions: {
+        watchQuery: {
+          fetchPolicy: 'no-cache'
+        },
+        query: {
+          fetchPolicy: 'no-cache'
+        },
+        mutate: {
+          fetchPolicy: 'no-cache'
+        }
+      }
+    };
+  }
+
+  /**
+   * @override
+   * @property {Object} options - set the api used at the service level rather than the app level
+   */
+  get options() {
+    const defaultOptions = super.options;
+
+    return Object.assign({}, defaultOptions, {
+      apiURL: this._buildURLPath(),
+      credentials: 'include'
+    });
+  }
+
+  /**
+   * @override
+   * @method link
+   * @returns {ApolloLink} Adds our own headers to the request sent at the end of the Apollo links
+   */
+  link() {
+    const httpLink = super.link();
+    const headersLink = setContext((_, context) => {
+      context.headers = Object.assign(context.headers || {}, {
+        'Content-Type': 'application/json',
+        clientId: 'UI'
+      });
+
+      return context;
+    });
+
+    return headersLink.concat(httpLink);
+  }
+
+  /**
+   * @method _buildURLPath
+   * @returns {String} complete url including host and namespace
+   */
+  _buildURLPath() {
+    const host = getHost();
```

Comment:
use `configHost` rather than `getHost` and pass options through here.

Reply:
I think we may have to come back to this due to the way that the apollo client is instantiated. It seems like it's not meant to change the api url after instantiating the service so it may be best to define a separate service for each data source. That could make it simpler if there are different configurations needed for each data source's requests as well.